fe2-android/docs/README.md

```diff
@@ -165,13 +165,61 @@ This is the current organization, but as this project evolves you should feel fr
 
 For more information on linking up together the user interface with the application state, please make sure you have a solid understanding of the [MVVM pattern](#application-design) and look at the `ViewModel` classes and the `LAORepository`.
 
+### Dependency Injection
 
-## Testing
+The project uses dependency injection to provide the different dependencies instances. If you don't know what it is, android provides a [simple guide](https://developer.android.com/training/dependency-injection).
+
+Hilt is used to do the injection. It is an annotation based frameworks. It is especially useful when writing tests because it allows the user to choose specifically what is injected and thus what module is mocked and how it is mocked.
+
+For detailed information, take a look at those :
+* [Manual dependency injection](https://developer.android.com/training/dependency-injection/manual) To take a look at what hilt does by himself.
+* [Dependency injection using Hilt](https://developer.android.com/training/dependency-injection/hilt-android) For an overview of the framework.
+* [Testing using Hilt](https://developer.android.com/training/dependency-injection/hilt-testing) To learn how to test
+
+#### Hilt quick start
+
+To provide an instance, you need to define it in a Module. A Module is a collection of rules on how to provide a dependency. The modules of the project are defined in the [di package](https://github.com/dedis/student_21_pop/tree/master/fe2-android/app/src/main/java/com/github/dedis/popstellar/di).
+
+You can either directly provide an instance with arguments or a binding to link a type to another. Providing an instance is done with a static function annotated with @Provides. Binding the types A -> B is done through an abstract method that returns A and takes B as argument.
+
+Example :
+In a project where we have the interface Database and its implementation DatabaseImpl we could find the module :
+```java
+@Module
+public abstract class DatabaseModule {
 
-For testing go to `Build Variants` and choose *mockDebug* as the `Active Build Variant`. 
+  @Binds
+  public abstract Database bindDatabase(DatabaseImpl impl);
+
+  @Provides
+  public static DatabaseImpl provideDatabaseImpl(NetworkService networkService) {
+    return new DatabaseImpl(networkService);
+  }
+}
+```
+
+If a module needs Database, Hilt will find the Binds rule that specifies it should look for a DatabaseImpl. Then it finds the Provides rule and can provide the actual instance of the Database object. Here, the `networkService` argument is provided by Hilt.
```

Comment:
Where does this `NetworkService` come from ? What does it refer to? Is it a Hilt resource or something you assume gets injected ? This needs to be clarified, ideally ahead of the example, otherwise the example will be even more confusing

Reply:
I made some changes regarding this, I hope it is more clear.